platforms/fabric/src/main/java/com/dfsek/terra/fabric/generation/BeardGenerator.java

```diff
@@ -110,11 +108,13 @@ public void generate(ChunkGenerator generator, WorldProperties worldProperties,
     }
     
     
+    @SuppressWarnings("RedundantCast") // Suppressed to prevent IDEs from removing these casts causing ClassCastExceptions
     public double calculateNoise(int x, int y, int z) {
         double noise = 0.0;
         
-        while (this.pieceIterator.hasNext()) {
-            StructurePiece structurePiece = this.pieceIterator.next();
+        for (Object __structurePiece: (Object[]) this.pieces.elements()) {
```

Comment:
Iterators, including iterators from concurrent implementations, are not designed to be used concurrently. This would cause inconsistencies between iterations. 
Also, iterators and enhanced loop with an `Iterable` are generally slower than array iteration.